pkgs/development/python-modules/apache-airflow/update-providers.py

```diff
@@ -0,0 +1,226 @@
+#! /usr/bin/env python3
+
+from itertools import chain
+import json
+import logging
+from pathlib import Path
+import os
+import re
+import subprocess
+import sys
+from typing import Dict, List, Optional, Set, TextIO
+from urllib.request import urlopen
+from urllib.error import HTTPError
+import yaml
+
+PKG_SET = "pkgs.python3Packages"
+
+# If some requirements are matched by multiple or no Python packages, the
+# following can be used to choose the correct one
+PKG_PREFERENCES = {
+    "dnspython": "dnspython",
+    "google-api-python-client": "google-api-python-client",
+    "psycopg2-binary": "psycopg2",
+    "requests_toolbelt": "requests-toolbelt",
+}
+
+# Requirements missing from the airflow provider metadata
+EXTRA_REQS = {
+    "sftp": ["pysftp"],
+}
+
+
+def get_version():
+    with open(os.path.dirname(sys.argv[0]) + "/default.nix") as fh:
+        # A version consists of digits, dots, and possibly a "b" (for beta)
+        m = re.search('version = "([\\d\\.b]+)";', fh.read())
+        return m.group(1)
```

Comment:
We should get the version via nix.

Reply:
The script largely copies from home-assistant's parse-requirements.py, and it has worked a long time like this.